single_kernel_mongo/managers/sharding.py

```diff
@@ -671,3 +675,124 @@ def drained(self, mongos_hosts: list[str], shard_name: str):
 
         self.state.unit_peer_data.drained = drained
         return drained
+
+    def cluster_password_synced(self) -> bool:
+        """Returns True if the cluster password is synced."""
+        # base case: not config-server
+        if not self.state.is_role(MongoDBRoles.SHARD):
+            return True
+
+        try:
+            # check our ability to use connect to mongos
+            config = self.state.mongos_config_for_user(
+                OperatorUser, set(self.state.app_peer_data.mongos_hosts)
+            )
+
+            with MongoConnection(config) as mongos:
+                mongos.get_shard_members()
+            # check our ability to use connect to mongod
+            with MongoConnection(self.state.mongo_config) as mongod:
+                mongod.get_replset_status()
+        except OperationFailure as e:
+            if e.code in [13, 18, 133]:
+                return False
+            raise
+        except ServerSelectionTimeoutError:
+            # Connection refused, - this occurs when internal membership is not in sync across the
+            # cluster (i.e. TLS + KeyFile).
+            return False
+
+        return True
+
+    def _is_added_to_cluster(self) -> bool:
+        """Returns true if the shard has been added to the clusted."""
+        if not self.state.config_server_name or not self.state.app_peer_data.mongos_hosts:
+            return False
+
+        try:
+            config = self.state.mongos_config_for_user(
+                OperatorUser, set(self.state.app_peer_data.mongos_hosts)
+            )
+            # check our ability to use connect to mongos
+            with MongoConnection(config) as mongos:
+                members = mongos.get_shard_members()
+        except OperationFailure as e:
+            if e.code in [13, 18, 133]:
+                return False
+            raise
+        except ServerSelectionTimeoutError:
+            # Connection refused, - this occurs when internal membership is not in sync across the
+            # cluster (i.e. TLS + KeyFile).
+            return False
+
+        return self.state.app_peer_data.replica_set in members
+
+    def _is_shard_aware(self) -> bool:
+        config = self.state.mongos_config_for_user(
+            OperatorUser, set(self.state.app_peer_data.mongos_hosts)
+        )
+        with MongoConnection(config) as mongo:
+            return mongo.is_shard_aware(self.state.app_peer_data.replica_set)
+
+    def skip_shard_status(self) -> bool:
```

Comment:
```suggestion
    def should_skip_shard_status(self) -> bool:
```
nit: bool functions should start with a question word